plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListener.java

```diff
@@ -0,0 +1,376 @@
+                                        .facets(
+                                                openLineage.newJobFacetsBuilder()
+                                                        .sql(openLineage.newSQLJobFacet(queryCreatedEvent.getMetadata().getQuery()))
+                                                        .build())
+                                        .build())
+                        .build();
+
+        return startEvent;
+    }
+
+    public OpenLineage.RunEvent getCompletedEvent(UUID runID, QueryCompletedEvent queryCompletedEvent)
+            throws IllegalAccessException
+    {
+        boolean failed = queryCompletedEvent.getMetadata().getQueryState().equals("FAILED");
+
+        OpenLineage.RunFacetsBuilder runFacetsBuilder = openLineage.newRunFacetsBuilder();
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_METADATA.getText(),
+                getTrinoMetadataFacet(queryCompletedEvent.getMetadata()));
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_QUERY_CONTEXT.getText(),
+                getTrinoQueryContextFacet(queryCompletedEvent.getContext()));
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_QUERY_STATISTICS.getText(),
+                getTrinoQueryStatisticsFacet(queryCompletedEvent.getStatistics()));
+
+        OpenLineage.RunEvent completedEvent =
+                openLineage.newRunEventBuilder()
+                        .eventType(
+                                failed
+                                        ? OpenLineage.RunEvent.EventType.FAIL
+                                        : OpenLineage.RunEvent.EventType.COMPLETE)
+                        .eventTime(queryCompletedEvent.getEndTime().atZone(ZoneId.of("UTC")))
+                        .run(openLineage.newRunBuilder().runId(runID).facets(runFacetsBuilder.build()).build())
+                        .job(
+                                openLineage.newJobBuilder()
+                                        .namespace(this.jobNamespace)
+                                        .name(queryCompletedEvent.getMetadata().getQueryId())
+                                        .facets(
+                                                openLineage.newJobFacetsBuilder()
+                                                        .sql(openLineage.newSQLJobFacet(queryCompletedEvent.getMetadata().getQuery()))
+                                                        .build())
+                                        .build())
+                        .inputs(buildInputs(queryCompletedEvent.getIoMetadata()))
+                        .outputs(buildOutputs(queryCompletedEvent.getIoMetadata()))
+                        .build();
+
+        return completedEvent;
+    }
+
+    private List<OpenLineage.InputDataset> buildInputs(QueryIOMetadata ioMetadata)
+    {
+        return ioMetadata.getInputs().stream().map(inputMetadata ->
+                openLineage.newInputDatasetBuilder()
+                        .namespace(getDatasetNamespace())
```

Comment:
Can we directly use `jobNamespace` here ?

Reply:
I've just aligned everything to `this.namespace`